Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Magic secret

## Developer changelog:
- Added extra conditions to magic perk selection to not give trait if magic is illegal
- reduced penalty multiplier for learning secret magic from 0 to 25 percent
- added secrets upon perk selection for application if magic is illegal
- added basic framework for events in the magic system
- added basic events for revelation of magic secrets
- removed all bonuses from trait perks
- ruler metaclass trait only applied for learning vanilla perks
- added custom loc for magic secret related insults


## Tests:


# How to test:
- add a perk in a tree which is illegal in your religion
- trait should not appear in your character screen
- switch to another character and use debug or other methods to reveal test characters secrets
- blackmail the test character
- switch back to test character and deny blackmail request
- event should pop up
- immediately switch to family member
- event should pop up
